baybe/searchspace/discrete.py

```diff
@@ -230,6 +231,122 @@ def discrete_parameter_factory(
+        with appropriate sum constraints. However, such an approach is inefficient
+        because the Cartesian product involved creates an exponentially large set of
+        candidates, most of which do not satisfy the simplex constraints and must be
+        subsequently be filtered out by the method.
+
+        By contrast, this method uses a shortcut that removes invalid candidates
+        already during the creation of parameter combinations, resulting in a
+        significantly faster construction.
+
+        Args:
+            parameters: The parameters to be used for building the subspace.
+                Parameters of type
+                :class:`baybe.parameters.numerical.NumericalDiscreteParameter`
+                enter the simplex construction while other subclasses of
+                :class:`baybe.parameters.base.DiscreteParameter` enter in form
+                of a Cartesian product.
+            total: The maximum sum of the parameter values defining the simplex size.
+            boundary_only: Flag determining whether to keep only parameter
+                configurations on the simplex boundary.
+            tolerance: Numerical tolerance used to validate the simplex constraint.
+
+        Raises:
+            ValueError: If the passed parameters are not suitable for a simplex
+                construction.
+
+        Returns:
+            The created simplex subspace.
+
+        Note:
+            The achieved efficiency gains can vary depending on the particular order in
+            which the parameters are passed to this method, as the configuration space
+            is built up incrementally from the parameter sequence.
+        """
+        # Separate parameter into numerical and other
+        numerical_parameters = [
+            p for p in parameters if isinstance(p, NumericalDiscreteParameter)
+        ]
+        other_parameters = [
+            p for p in parameters if not isinstance(p, NumericalDiscreteParameter)
+        ]
+
+        # Validate non-negativity
+        min_values = [min(p.values) for p in numerical_parameters]
+        if not (min(min_values) >= 0.0):
+            raise ValueError(
+                f"All parameters passed to '{cls.from_simplex.__name__}' "
+                f"must have non-negative values only."
+            )
+
+        def drop_invalid(df: pd.DataFrame, total: float, boundary_only: bool) -> None:
+            """Drop rows that violate a specified simplex constraint.
+
+            Args:
+                df: The dataframe whose rows should satisfy the simplex constraint.
+                total: The maximum row sum defining the simplex size.
+                boundary_only: Flag to control if the points represented by the rows
+                    may lie inside the simplex or on its boundary only.
+            """
+            row_sums = df.sum(axis=1)
+            if boundary_only:
+                rows_to_drop = row_sums[
```

Comment:
Good point. Made it even more explicit by calling it `locs_to_drop` because all "idx"-based naming is also always somewhat ambiguous in dataframes (dammit pandas!!!)